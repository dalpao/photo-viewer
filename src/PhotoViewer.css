.myClass {
    color:aqua;
}

.title {
    color: aqua;
}

.img-gallery ul {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    list-style: none;
    justify-content: center;
    width: 970px;
    padding: 0;
}

.img-gallery ul li {
    width: 200px;
    margin-right: 10px;
}
.img-gallery img {
    width: 194px;
    height: 119.375px;
    border-radius: 5%;
    cursor: pointer;
}

.img-gallery img:hover {
    border: 3px solid  #eb9e34
}

/* .img-display img {
    width: 970px;
} */

.image {
    width: 830px;
    border-radius: 5%;
    border: 3px solid  #eb9e34
}