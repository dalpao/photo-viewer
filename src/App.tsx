import { useState } from 'react'
import './App.css'
import { PhotoViewer } from './PhotoViewer'
import { ImageSelector } from './ImageSelector'



function App() {
    const [displayUrl, setUrl] = useState('');


  return (
      <div>
          <h1>React Photo Viewer</h1>
          <PhotoViewer name='Mary' url={displayUrl}/>
          <ImageSelector changeImg ={setUrl}/>
      </div>
  );
}


export default App
