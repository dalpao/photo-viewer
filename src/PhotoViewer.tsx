import './PhotoViewer.css'

export function PhotoViewer(props: {name: string, url: string}) {    
    return (                
        <div className="myClass">               
           <h1 className="title"> Hello {props.name}! </h1>
           <img className="image" src={props.url} alt='The selected view'/>
        </div>
    );
}
