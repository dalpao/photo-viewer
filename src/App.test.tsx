
import { render } from '@testing-library/react';
import App from './App';
import { imageUrls } from './imageUrls'
import { PhotoViewer } from './PhotoViewer'
import * as renderer from 'react-test-renderer';
// import { ImageSelector } from './ImageSelector';


test('renders Hello world text', () => {
  const { getByText } = render(<App />);
  const textElement = getByText(/Hello Mary/i);
  expect(textElement).toBeInTheDocument();
});

test('imgageUrl generation code correctly', () => {
    const firstImgUrl = 'https://picsum.photos/id/600/1600/900.jpg';

    const result = imageUrls[0];

    expect(result).toBe(firstImgUrl);
  });

  test('broken images are not included', () => {

    const brokenImgUrls = [
        'https://picsum.photos/id/601/1600/900.jpg', 
        'https://picsum.photos/id/624/1600/900.jpg',
        'https://picsum.photos/id/632/1600/900.jpg',
        'https://picsum.photos/id/636/1600/900.jpg',
        'https://picsum.photos/id/644/1600/900.jpg',
        'https://picsum.photos/id/647/1600/900.jpg'
    ];

 
    brokenImgUrls.forEach(url => {
        // expect(imageUrls).not.toContain(url);
        expect(imageUrls.includes(url)).toBe(false);
    });
  });

  describe("PhotoViewer Component", () => {
    it ("Should match snapshot with text and image", async() => {
        const testName = 'Mary';
        const testUrl = 'https://picsum.photos/id/600/1600/900.jpg'
        const tree = renderer.create(<PhotoViewer name={testName} url={testUrl} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
  })


   







   
   