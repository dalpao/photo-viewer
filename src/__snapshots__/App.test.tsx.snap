// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoViewer Component Should match snapshot with text and image 1`] = `
<div
  className="myClass"
>
  <h1
    className="title"
  >
     Hello 
    Mary
    ! 
  </h1>
  <img
    alt="The selected view"
    className="image"
    src="https://picsum.photos/id/600/1600/900.jpg"
  />
</div>
`;
